AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/QtPyScriptCanvasVariableManager.py

```diff
@@ -40,39 +41,110 @@ def __init__(self, sc_editor):
 
-        param new_variable_type: the type of variable you want to create provided as a string. Is case-sensitive!
-
-        returns: None
+        returns None
         """
-        self.__validate_new_variable(new_variable_type)
-
         add_new_variable_button = self.variable_manager.findChild(QtWidgets.QPushButton, ADD_BUTTON_QT)
         add_new_variable_button.click()  # Click on Create Variable button
 
         helper.wait_for_condition((
             lambda: self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT) is not None),
             WAIT_TIME_SEC_3)
 
+    def __select_variable_type_from_create_menu(self, new_variable_type: str) -> None:
+        """
+        helper function for adding a new variable. selects the desired variable type from the menu that renders after
+        the create variable button is clicked. waits for the UI to respond after making a selection
+
+        params: The type of variable to create as a string. ie "Boolean"
+
+        returns none
+        """
+
         # Select variable type
         table_view = self.variable_manager.findChild(QtWidgets.QTableView, VARIABLE_PALETTE_QT)
         variable_entry = pyside_utils.find_child_by_pattern(table_view, new_variable_type)
 
-        # Click on it to create variable
+        # Click on it to create variable and wait 1 second for the UI to respond
         pyside_utils.item_view_index_mouse_click(table_view, variable_entry)
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
+
+    def __set_new_variable_name(self, new_variable_name: str):
+
+        """
+        helper function for adding a new variable. sets the new variable's name. waits for the UI to respond afterwards
+
+        returns Qwidget with the graph variables that have been created
+        """
+
+        # Set the new variable's name. Wait 1 second for the UI to respond
+        graph_vars_page = self.variable_manager.findChild(QtWidgets.QWidget, GRAPH_VARIABLES_PAGE_QT)
+        line_edit = graph_vars_page.findChild(QtWidgets.QLineEdit, "")
+        line_edit.insert(new_variable_name)
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
+
+        return graph_vars_page
+
+    def __get_variable_selection_index(self, graph_vars_page: QtWidgets.QWidget):
+        """
+        helper function for adding a new variable. gets the index object from the created graph variables widget then
+        emulates clicking somewhere else on the UI to remove focus from the graph variables . waits for the UI to
+        respond afterwards.
+
+        returns selection Qobject
+        """
+        selection_model = graph_vars_page.findChild(QtCore.QItemSelectionModel)
+        selection = selection_model.selection()
+        selection_model.clear()
+        helper.wait_for_condition(lambda: True is False, WAIT_TIME_SEC_1)
```

Comment:
Alarming wait.

Reply:
replaced this with a wait for condition and a check to see if the selection model does not have a selection